platform/darwin/src/MGLFeature_Private.h

```diff
@@ -42,4 +42,10 @@ NS_DICTIONARY_OF(NSString *, id) *NSDictionaryFeatureForGeometry(NSDictionary *g
 
 @end
 
+@protocol MGLFeatureCollectionPrivate <NSObject>
```

Comment:
Hidden protocols are dangerous because they make it easy to write code that crashes on unrecognized selectors. MGLFeaturePrivate is necessary because MGLFeature is a protocol, but if MGLShapeCollectionFeature is the only class that would conform to this hidden protocol, it’d be safer to declare a private extension on MGLShapeCollection specifically.